packages/client/src/components/inspector/InspectorStateField.vue

```diff
@@ -112,15 +112,15 @@ const { editingType, editing, editingText, toggleEditing, nodeId } = useStateEdi
 
 watch(() => editing.value, (v) => {
   if (v) {
-    const { value } = rawValue.value
-    editingText.value = toEdit(value)
+    const { value } = raw.value
+    editingText.value = toEdit(value, raw.value.customType)
   }
   else {
     editingText.value = ''
   }
 })
 
-function submit(dataType: string) {
```

Comment:
IMO, The `dataType` is not necessary.
It is passed to the component `EditInput`, and emits back to the parent `InspectorStateField` while unchanged.